lib/compat/wordpress-6.5/blocks.php

```diff
@@ -45,3 +45,82 @@ function gutenberg_register_metadata_attribute( $args ) {
 	return $args;
 }
 add_filter( 'register_block_type_args', 'gutenberg_register_metadata_attribute' );
+
+require_once __DIR__ . '/block-bindings/index.php';
+
+if ( ! function_exists( 'gutenberg_process_block_bindings' ) ) {
+	/**
+	 * Process the block bindings attribute.
+	 *
+	 * @param string   $block_content Block Content.
+	 * @param array    $block Block attributes.
+	 * @param WP_Block $block_instance The block instance.
+	 */
+	function gutenberg_process_block_bindings( $block_content, $block, $block_instance ) {
+
+		// Allowed blocks that support block bindings.
+		// TODO: Look for a mechanism to opt-in for this. Maybe adding a property to block attributes?
+		$allowed_blocks = array(
+			'core/paragraph' => array( 'content' ),
+			'core/heading'   => array( 'content' ),
+			'core/image'     => array( 'url', 'title', 'alt' ),
+			'core/button'    => array( 'url', 'text' ),
+		);
+
+		// If the block doesn't have the bindings property or isn't one of the allowed block types, return.
+		if ( ! isset( $block['attrs']['metadata']['bindings'] ) || ! isset( $allowed_blocks[ $block_instance->name ] ) ) {
+			return $block_content;
+		}
+
+		// Assuming the following format for the bindings property of the "metadata" attribute:
+		//
+		// "bindings": {
+		//   "title": {
+		//     "source": {
+		//       "name": "post_meta",
+		//       "attributes": { "value": "text_custom_field" }
+		//     }
+		//   },
+		//   "url": {
+		//     "source": {
+		//       "name": "post_meta",
+		//       "attributes": { "value": "text_custom_field" }
+		//     }
+		//   }
+		// }
+		//
```

Comment:
For the backport to Core, multiline comments are with /* */

```suggestion
		/*
		 * Assuming the following format for the bindings property of the "metadata" attribute:
		 *
		 * "bindings": {
		 *   "title": {
		 *     "source": {
		 *       "name": "post_meta",
		 *       "attributes": { "value": "text_custom_field" }
		 *     }
		 *   },
		 *   "url": {
		 *     "source": {
		 *       "name": "post_meta",
		 *       "attributes": { "value": "text_custom_field" }
		 *     }
		 *   }
		 * }
		 */
```